packages/frontend2/src/app/(new)/(other)/scaling/tvl/_components/scaling-tvl-chart.tsx

```diff
@@ -0,0 +1,274 @@
+      range={[rangeStart, rangeEnd]}
+      useLogScale={scale === 'log'}
+      renderHoverContents={(data) => <ChartHover {...data} currency={unit} />}
+    >
+      <section className="flex flex-col gap-4">
+        <Header
+          unit={unit}
+          value={lastValue}
+          change={change}
+          range={timeRange}
+        />
+        <ChartTimeRangeControls
+          value={timeRange}
+          setValue={setTimeRange}
+          options={[
+            { value: '7d', label: '7D' },
+            { value: '30d', label: '30D' },
+            { value: '90d', label: '90D' },
+            { value: '180d', label: '180D' },
+            { value: '1y', label: '1Y' },
+            { value: 'max', label: 'MAX' },
+          ]}
+          range={[rangeStart, rangeEnd]}
+        />
+        <Chart />
+        <UnitAndScaleControls
+          unit={unit}
+          scale={scale}
+          setUnit={setUnit}
+          setScale={setScale}
+        />
+      </section>
+    </ChartProvider>
+  )
+}
+
+function ChartHover(props: {
+  timestamp: number
+  canonical: number
+  external: number
+  native: number
+  ethPrice: number
+  currency: 'usd' | 'eth'
+}) {
+  const divideBy = props.currency === 'usd' ? 100 : props.ethPrice
+  const total = formatNumber(
+    (props.canonical + props.external + props.native) / divideBy,
+    2,
+  )
+  const values = [
+    {
+      title: 'Canonical',
+      value: props.canonical,
+      className: 'bg-purple-100 dark:bg-purple-100',
+    },
+    {
+      title: 'External',
+      value: props.external,
+      className: 'bg-yellow-200 dark:bg-yellow-200',
+    },
+    {
+      title: 'Native',
+      value: props.native,
+      className: 'bg-pink-100 dark:bg-pink-100',
+    },
+  ]
+  return (
+    <div className="flex flex-col gap-1">
+      <div className="whitespace-nowrap">
+        {formatTimestamp(props.timestamp, {
+          mode: 'datetime',
+        })}
+      </div>
+      <div className="flex w-full items-center justify-between gap-2">
+        <span className="text-sm text-gray-700 dark:text-gray-50">
+          Total TVL
+        </span>
+        {props.currency === 'usd' ? '$' : 'Ξ'}
+        {total}
+      </div>
+      <hr className="w-full border-gray-200 dark:border-gray-650 md:border-t" />
+      <div>
+        {values.map(
+          (v, i) =>
+            v.value > 0 && (
+              <div
+                key={i}
+                className="flex items-center justify-between gap-x-6"
+              >
+                <span className="flex items-center gap-1">
+                  <div
+                    role="img"
+                    aria-label="Square icon"
+                    className={cn('size-3 rounded', v.className)}
+                  ></div>
+                  <span>{v.title}</span>
+                </span>
+                <span className="font-semibold">
+                  {props.currency === 'usd' ? '$' : 'Ξ'}
+                  {formatNumber(v.value / divideBy)}
```

Comment:
`formatCurrency`

why do we need to divide it?

Reply:
Divide is needed because the value is passed in cents or us dollars. The divider is either 100 or ethPrice (which is also in cents, so /100 is no longer needed).